Common/Data/Market/Greeks.cs

```diff
@@ -0,0 +1,195 @@
+        /// <para>
+        /// Vega measures the rate of change of the option value with respect to changes in
+        /// the underlying's volatility. (∂V/∂σ)
+        /// </para>
+        /// </summary>
+        public decimal Vega
+        {
+            get
+            {
+                return _vega.Value;
+            }
+            private set
+            {
+                _vega = new Lazy<decimal>(() => value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the theta.
+        /// <para>
+        /// Theta measures the rate of change of the option value with respect to changes in
+        /// time. This is commonly known as the 'time decay.' (∂V/∂τ)
+        /// </para>
+        /// </summary>
+        public decimal Theta
+        {
+            get
+            {
+                return _theta.Value;
+            }
+            private set
+            {
+                _theta = new Lazy<decimal>(() => value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the rho.
+        /// <para>
+        /// Rho measures the rate of change of the option value with respect to changes in
+        /// the risk free interest rate. (∂V/∂r)
+        /// </para>
+        /// </summary>
+        public decimal Rho
+        {
+            get
+            {
+                return _rho.Value;
+            }
+            private set
+            {
+                _rho = new Lazy<decimal>(() => value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the lambda.
+        /// <para>
+        /// Lambda is the percentage change in option value per percentage change in the
+        /// underlying's price, a measure of leverage. Sometimes referred to as gearing.
+        /// (∂V/∂S ✕ S/V)
+        /// </para>
+        /// </summary>
+        public decimal Lambda
+        {
+            get
+            {
+                return _lambda.Value;
+            }
+            private set
+            {
+                _lambda = new Lazy<decimal>(() => value);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new default instance of the <see cref="Greeks"/> class
+        /// </summary>
+        public Greeks() 
+            : this(0m, 0m, 0m, 0m, 0m, 0m)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Greeks"/> class
+        /// </summary>
+        public Greeks(decimal delta, decimal gamma, decimal vega, decimal theta, decimal rho, decimal lambda)
+        {
+            Delta = delta;
+            Gamma = gamma;
+            Vega = vega;
+            Theta = theta;
+            Rho = rho;
+            Lambda = lambda;
+        }
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Greeks"/> class
+        /// </summary>
+        public Greeks(Func<decimal> delta, Func<decimal> gamma, Func<decimal> vega, Func<decimal> theta, Func<decimal> rho, Func<decimal> lambda)
```

Comment:
yes, on demand computations. Afraid of expensive computations quickly growing on options side.